sqlccl: fix a data race in TestTenantTempTableCleanup

knz opened the pull request:
Fixes #111028.
Epic: CRDB-6671

Release note: None

knz committed:
sqlccl: fix a data race in TestTenantTempTableCleanup

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/111054)
<!-- Reviewable:end -->

knz committed:
settings: clarify missharing of cluster.Settings

Prior to this patch, if a test incorrectly shared a `cluster.Settings`
object across servers, there was a chance the error would pop up as a
race condition. We really want this to pop up as an error instead.

This patch ensures that the race detector is always happy so a clean
error is always reported in misuses.

Release note: None

yuzefovich reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> Thanks!

> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111054#-:-NescAz9DpmuAZATM9xG:b-g2a5fb)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @DrewKimball)



<!-- Sent from Reviewable.io -->

knz commented:
TFYR!

bors r=yuzefovich

yuzefovich commented:
Looks like bors crashed.

bors r=yuzefovich
